lzy/allocator/src/main/java/ai/lzy/allocator/alloc/impl/kuber/KuberVmAllocator.java

```diff
@@ -397,68 +356,76 @@ public Result deallocate(Vm vm) {
     }
 
     @Override
-    public List<VmEndpoint> getVmEndpoints(String vmId, @Nullable TransactionHandle transaction) {
-        final List<VmEndpoint> hosts = new ArrayList<>();
+    public Vm updateAllocatedVm(Vm vm, @Nullable TransactionHandle tx) {
+        var meta = requireNonNull(vm.allocateState().allocatorMeta());
 
-        final var meta = withRetries(
-            defaultRetryPolicy(),
-            LOG,
-            () -> vmDao.getAllocatorMeta(vmId, transaction),
-            ex -> new RuntimeException("Database error: " + ex.getMessage(), ex));
-
-        if (meta == null) {
-            throw new RuntimeException("Cannot get allocator metadata for vmId " + vmId);
-        }
-
-        final var clusterId = meta.get(CLUSTER_ID_KEY);
-        final var credentials = clusterRegistry.getCluster(clusterId);
-        final var ns = meta.get(NAMESPACE_KEY);
-        final var podName = meta.get(POD_NAME_KEY);
+        final var clusterId = requireNonNull(meta.get(CLUSTER_ID_KEY));
+        final var credentials = requireNonNull(clusterRegistry.getCluster(clusterId));
+        final var ns = requireNonNull(meta.get(NAMESPACE_KEY));
+        final var podName = requireNonNull(meta.get(POD_NAME_KEY));
 
         try (final var client = k8sClientFactory.build(credentials)) {
             final var pod = getVmPod(ns, podName, client);
-            if (pod != null) {
-                final var nodeName = pod.getSpec().getNodeName();
-                final var node = client.nodes().withName(nodeName).get();
-
-                final var providerId = node.getSpec() != null ? node.getSpec().getProviderID() : null;
-                final var instanceId = providerId != null && providerId.startsWith("yandex://")
-                    ? providerId.substring("yandex://".length())
-                    : null;
-
-                LOG.info("VM {} is allocated at POD {} on node {} ({})", vmId, podName, nodeName, providerId);
-
-                for (final var address : node.getStatus().getAddresses()) {
-                    final var type = switch (address.getType().toLowerCase()) {
-                        case "hostname" -> VmEndpointType.HOST_NAME;
-                        case "internalip" -> VmEndpointType.INTERNAL_IP;
-                        case "externalip" -> VmEndpointType.EXTERNAL_IP;
-                        default -> throw new RuntimeException("Undefined type of node address: " + address.getType());
-                    };
-                    hosts.add(new VmEndpoint(type, address.getAddress()));
-                }
-
-                meta.put(NODE_NAME_KEY, nodeName);
-                if (instanceId != null) {
-                    meta.put(NODE_INSTANCE_ID_KEY, instanceId);
-                }
-            } else {
+            if (pod == null) {
                 throw new RuntimeException("Cannot get pod with name " + podName + " to get addresses");
             }
+
+            final var nodeName = pod.getSpec().getNodeName();
```

Comment:
Not sure if it's okay to expect that nodeName will be not null. For example, pod can be in scheduling state, waiting for free nodes.

Reply:
this method (updateAllocatedVm) is called _after_ allocation only, on callback from allocator agent

Reply:
Oh, now I see. But it's would be better, if exception is different from simple NPE